java/core/src/test/java/sleeper/core/statestore/FilesReportTestHelper.java

```diff
@@ -21,60 +21,134 @@
     }
 
+    /**
+     * Creates a report with the specified files referenced in partitions.
+     *
+     * @param  files the file references
+     * @return       the report
+     */
     public static AllReferencesToAllFiles activeFiles(FileReference... files) {
         return activeFilesReport(DEFAULT_UPDATE_TIME, List.of(files));
     }
 
+    /**
+     * Creates a report with the specified files referenced in partitions. All files will be given the same last update
+     * time.
+     *
+     * @param  updateTime the update time
+     * @param  files      the file references
+     * @return            the report
+     */
     public static AllReferencesToAllFiles activeFilesReport(Instant updateTime, FileReference... files) {
         return activeFilesReport(updateTime, List.of(files));
     }
 
+    /**
+     * Creates a report with the specified files referenced in partitions. All files will be given the same last update
+     * time.
+     *
+     * @param  updateTime the update time
+     * @param  references the file references
+     * @return            the report
+     */
     public static AllReferencesToAllFiles activeFilesReport(Instant updateTime, List<FileReference> references) {
-        return new AllReferencesToAllFiles(AllReferencesToAFile
-                .newFilesWithReferences(references.stream(), updateTime)
-                .collect(Collectors.toUnmodifiableList()), false);
+        return new AllReferencesToAllFiles(
+                AllReferencesToAFile.newFilesWithReferences(references.stream())
+                        .map(file -> file.withCreatedUpdateTime(updateTime))
+                        .collect(Collectors.toUnmodifiableList()),
+                false);
     }
 
+    /**
+     * Creates a report with specified files referenced in partitions, and files with no references.
+     *
+     * @param  activeFiles     the file references
+     * @param  readyForGCFiles the filenames with no references
+     * @return                 the report
+     */
     public static AllReferencesToAllFiles activeAndReadyForGCFiles(List<FileReference> activeFiles, List<String> readyForGCFiles) {
         return activeAndReadyForGCFilesReport(DEFAULT_UPDATE_TIME, activeFiles, readyForGCFiles);
     }
 
+    /**
+     * Creates a report with specified files referenced in partitions, and files with no references.
+     *
+     * @param  activeFiles     the file references
+     * @param  readyForGCFiles the filenames with no references
+     * @return                 the report
+     */
     public static AllReferencesToAllFiles activeAndReadyForGCFilesReport(
             Instant updateTime, List<FileReference> activeFiles, List<String> readyForGCFiles) {
         return new AllReferencesToAllFiles(activeAndReadyForGCFiles(updateTime, activeFiles, readyForGCFiles), false);
     }
 
+    /**
+     * Creates a report with the specified files with no references in partitions.
```

Comment:
I dont think we need the ` in partitions` bit here? There won't be any references to the files in any partition.

Reply:
Done